include/mimick/mock.h

```diff
@@ -46,22 +47,26 @@ struct mmk_params {
     struct mmk_params *next;
 };
 
-#if defined(__GNUC__) && !defined(__clang__)
+# if defined(__GNUC__) && !defined(__clang__)
 // Force bit-field alignment to circumvent a bug in GCC
-// versions 6, 7, and 8, and a fixup notice on GCC 9.
+// versions 6, 7, and 8, and a ABI fix notice in GCC 9.
 // See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88469.
-// For 32 bit machines in which bit-field alignment matters,
-// this will ensure correct alignment. For 64 bit machines
-// in which bit-field alignment matters, we can reasonably
-// expect the compiler to tighten up alignment.
-struct __attribute__((aligned(4))) mmk_mock_options {
-#else
+#  if UINTPTR_MAX == UINT64_MAX
+#   define BITFIELD_ALIGNMENT __attribute__((aligned(8)))
+#  elif UINTPTR_MAX == UINT32_MAX
+#   define BITFIELD_ALIGNMENT __attribute__((aligned(4)))
+#  endif
+# else
+#  define BITFIELD_ALIGNMENT
+# endif
+
 struct mmk_mock_options {
-#endif
-    unsigned sentinel_  : 1;
-    unsigned noabort    : 1;
+  unsigned BITFIELD_ALIGNMENT sentinel_  : 1;
+  unsigned BITFIELD_ALIGNMENT noabort    : 1;
```

Comment:
I'm OK with this change, but I'll point out a couple of things:

1.  According to https://gcc.gnu.org/onlinedocs/gcc-9.3.0/gcc/Common-Type-Attributes.html, you can use `__attribute__((aligned))` and it will use the "maximum alignment" for the target.  That probably means either 8 or 16 bytes, which increases the structure size, but also gets rid of all of the `#if defined` stuff above.
1.  If the size of this structure doesn't matter, can we just sidestep the whole issue and don't use bitfields?  That is, redefine this structure to be:

```
struct mmk_mock_options {
  uint32_t sentinel_;
  uint32_t noabort;
};
```

Reply:
> [...] you can use __attribute__((aligned)) and it will use the "maximum alignment" for the target. That probably means either 8 or 16 bytes, which increases the structure size, but also gets rid of all of the #if defined stuff above.

Fair point. I don't have a strong opinion. Do you?

> If the size of this structure doesn't matter, can we just sidestep the whole issue and don't use bitfields?

I was purposefully trying to not touch the API too much just to minimize friction when we attempt to upstream this.

Reply:
> Fair point. I don't have a strong opinion. Do you?

Not a strong opinion, but if we're trying to keep differences to upstream low, then I think having a single line change of adding the `__attribute__((aligned))` is simpler than adding all of the `#ifdef` stuff.  I'll approve anyway and let you decide which way to go.

Reply:
Alright, though we do need alignment enforced for each member field. See https://github.com/ros2/Mimick/pull/11/commits/be761fe4f0a3a009b2ac8468d4588d184b374bbe.